pennylane/devices/default_clifford.py

```diff
@@ -519,58 +692,322 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
+        if not tgt_states.shape[1]:
+            raise ValueError("Cannot set an empty list of target states.")
+
+        if len(meas_wires) < tgt_states.shape[1]:
+            tgt_states = []
+            for state in self._prob_states:
+                if list(state[meas_wires]) not in tgt_states:
+                    tgt_states.append(list(state[meas_wires]))
+            tgt_states = np.array(tgt_states)
+        else:
+            meas_wires = meas_wires[: tgt_states.shape[1]]
+
+        # Maintain a representaiton of basis states to build a visit-array
+        tgt_integs = np.array([int("".join(map(str, tgt_state)), 2) for tgt_state in tgt_states])
+
+        # Iterate over the required basis states and for each of them compute the probability
+        # If an impossible branch occur keep a note of it via a visit-array
+        # Worst case complexity O(B * M * N), where N is #measured_qubits and B is basis states.
+        visited_probs = []
+        prob_res = np.ones(tgt_states.shape[0])
+        for tgt_index, (tgt_integ, tgt_state) in enumerate(zip(tgt_integs, tgt_states)):
+            if tgt_integ in visited_probs:
+                continue
+            prob_sim = circuit_simulator.copy()
+            for idx, wire in enumerate(meas_wires):
+                expectation = prob_sim.peek_z(wire)
+                # (Eig --> Res) | -1 --> 1 | 1 --> 0 | 0 --> 0 / 1 |
+                outcome = int(0.5 * (1 - expectation))
+                if not expectation:
+                    prob_res[tgt_index] /= 2.0
+                else:
+                    nope_idx = (
+                        np.where(
+                            np.squeeze(np.all(tgt_states[:, :idx] == tgt_state[:idx], axis=-1))
+                            & tgt_states[:, idx]
+                            != outcome
+                        )[0]
+                        if idx
+                        else np.where(tgt_states[:, idx] != outcome)[0]
+                    )
+                    nope_idx = np.setdiff1d(nope_idx, visited_probs)
+                    prob_res[nope_idx] = 0.0
+                    visited_probs.extend(tgt_integs[nope_idx])
+
+                    if tgt_state[idx] != outcome:
+                        prob_res[tgt_index] = 0.0
+                        break
+
+                prob_sim.postselect_z(wire, desired_value=tgt_state[idx])
+            visited_probs.append(tgt_integ)
+        return prob_res
+
+    def _sample_expectation(self, meas, stim_circuit, shots, seed):
+        """Measure the expectation value with respect to samples from simulator device."""
+        # Get the observable for the expectation value measurement
+        meas_op = meas.obs
+        if meas_op is None:  # pragma: no cover
+            meas_op = qml.prod(
+                *[
+                    qml.PauliZ(idx)
+                    for idx in (meas.wires if meas.wires else range(stim_circuit.num_qubits))
+                ]
+            )
```

Comment:
shouldn't be possible, expval needs an observable. also, you still use `meas.obs` in the next line (which is fine), so we can just not define `meas_op` here
```suggestion
```

Reply:
I had added them to ensure operability if someone (possibly a `dev`) uses these internal methods directly for their workflow with `ExpectationMP` directly. 😓